runtime/stdlib/account.go

```diff
@@ -3747,6 +3853,18 @@ func newAuthAccountAccountCapabilitiesForEachControllerFunction(
 				panic(errors.NewUnreachableError())
 			}
 
+			// Prevent mutations (record/unrecord) to account capability controllers
+			// for this address during iteration
+
+			iterations := inter.SharedState.AccountCapabilityControllerIterations
+			iterations[address]++
```

Comment:
Hypothetically could we generalize this to being a map for all iterations over each different domain for each address? This would allow us to combine the storage iteration mutation and the capability controller iteration mutation into the same logic.

Reply:
Account Cap Cons do not have a path, but we can maybe use `common.PathDomainUnknown`

Reply:
Good idea! Simplified this in e5104616b035aaf4f0f9bf1e83252eca58c4a91a